consensus/beacon/consensus.go

```diff
@@ -426,23 +426,6 @@ func (beacon *Beacon) FinalizeAndAssemble(chain consensus.ChainHeaderReader, hea
 			panic("invalid tree type")
 		}
 		if okpre && okpost {
-			// TODO: see if this can be captured at the witness
-			// level, like it used to.
-			for _, key := range keys {
-				// WORKAROUND: the post trie would normally not
-				// need to be searched for keys, as all of them
-				// were resolved during block execution.
-				// But since the prefetcher isn't currently used
-				// with verkle, the values that are read but not
-				// written to, are not resolved as they are read
-				// straight from the snapshot. They must be read
-				// in order to build the proof.
-				_, err = vtrpost.GetWithHashedKey(key)
-				if err != nil {
-					panic(err)
-				}
-			}
```

Comment:
This shouldn't be needed from the previous PR that we properly use the `resolver` in the `GetElementsForProof`. Now, this is even more true since we don't do that for proof generation so this _should_ be safe.

I can't test it directly since this code path only works in real proof generation, so let's be aware the next time we update nodes in Kaustinen.